src/lib/vault.py

```diff
@@ -0,0 +1,88 @@
+import boto3
+import hvac
+import logging
+
+from hvac.api.auth_methods import Kubernetes
+
+from lib.export_backup import write_file
+
+
+class Vault:
+  ''' Interface with Hashicorp Vault '''
+  def __init__(
+    self,
+    addr,
+    auth_method,
+    token=None,
+    role=None,
+    kvv2_mount_point=None,
+    path=None,
+    sanitizer=None
+  ):
+    self._client = hvac.Client(url=addr)
+    self._mount_point = kvv2_mount_point
+    self._path = path
+    # a function for sanitizing a value before setting it in Vault
+    # this is org-specific
+    self._sanitizer = sanitizer
+    self._login(auth_method, token, role)
+
+  def _login(self, auth_method, token=None, role=None):
```

Comment:
Passing a project as a `role` here is a little confusing—It seems each project (e.g. gravity) has its own 'role' in Vault, and that role has its own list of secrets. Is that correct?

Reply:
Correct. I have to update our [K8S/Vault integration doc](https://www.notion.so/artsy/Integrate-Kubernetes-with-Vault-922ee8d71ac04507a5ad38b9003ca014). It's outdated - still talking about ESO.

Reply:
https://artsy.slack.com/archives/CA8SANW3W/p1728579560810289